bus/src/main/java/org/hawkular/metrics/component/publish/InsertedDataSubscriber.java

```diff
@@ -64,24 +71,35 @@
 
     private Subscription subscription;
 
+    @Inject
+    private PublishCommandTable publish;
+
+    @Inject
+    @Configurable
+    @ConfigurationProperty(METRICS_PUBLISH_PERIOD)
+    private String publishPeriodProperty;
+    private int publishPeriod;
+
+    private final List<Metric> pendingMetrics = new ArrayList<>();
+
+    private final Timer publishTimer = new Timer("PublishTimer");
+    private PublishTask publishTask;
+
     public void onMetricsServiceReady(@Observes @ServiceReady ServiceReadyEvent event) {
         Observable<List<Metric<?>>> events = event.getInsertedData().buffer(50, TimeUnit.MILLISECONDS, 100)
                 .filter(list -> !list.isEmpty())
                 .onBackpressureBuffer()
                 .observeOn(Schedulers.io());
         subscription = events.subscribe(list -> list.forEach(this::onInsertedData));
+        publishPeriod = getPublishPeriod();
+        publishTask = new PublishTask();
+        publishTimer.schedule(publishTask, 1000, publishPeriod);
     }
 
     private void onInsertedData(Metric<?> metric) {
         log.tracef("Inserted metric: %s", metric);
-        if (metric.getMetricId().getType() == AVAILABILITY) {
-            @SuppressWarnings("unchecked")
-            Metric<AvailabilityType> avail = (Metric<AvailabilityType>) metric;
-            publishAvailablility(avail);
-        } else {
-            @SuppressWarnings("unchecked")
-            Metric<? extends Number> numeric = (Metric<? extends Number>) metric;
-            publishNumeric(numeric);
+        if (publish.isPublished(metric.getMetricId())) {
```

Comment:
Also, since RxJava is used here otherwise.. this code should follow RxJava patterns.

Basically, instead of this codeblock, use something like

```java
metrics.filter(m -> publish.isPublished(m.getMetricId())).map(pendingMetrics:add)
```

Or make it even prettier with some more reactive handling.